Updated data wrangle & sea lice to refer to data package files

jclgan opened the pull request:
Edits to data_wrangle.rmd: Instead of {r setup} and {r sealice} referring to the hakaisalmon R package, they now refer to the .csv files in https://github.com/HakaiInstitute/jsp-data-packages.

Applied `filter(yday < 190)` to 2019 field data so it is restricted to the same sampling date range as 2015-2018 data.

Due to sea louse data in the data package having different table structure than in hakaisalmon, script has been updated to produce the same output format.

No changes have been made to {r sealice bootstrapping} as Brett should review the code.

jclgan committed:
Updated data wrangle & sea lice to refer to data package files

jclgan commented:
Note that since the data package files are stored in another repository/R project, I had to use absolute filepaths instead of here::here(). Is there a way to call on the files in the data package, or would we need to version the package and read in the .csvs to store in this project repo?

Br-Johnson commented on scripts/data_wrangle.Rmd:
```diff
@@ -51,25 +50,30 @@ survey_data$sampling_week <- recode_factor(survey_data$sampling_week,
 # Import data from 2015-2018
-survey_seines_2015_2018 <- hakaisalmon::survey_seines
-
+survey_seines_2015_2018 <- left_join(read_csv("../../jsp-data-packages/data/survey_data.csv"), read_csv("../../jsp-data-packages/data/seine_data.csv")) %>% 
```
You should use the http url instead here in this case. 
```suggestion
survey_seines_2015_2018 <- left_join(read_csv("https://raw.githubusercontent.com/HakaiInstitute/jsp-data/master/data/survey_data.csv"), read_csv("https://raw.githubusercontent.com/HakaiInstitute/jsp-data/master/data/seine_data.csv")) %>% 
```

Br-Johnson commented on scripts/data_wrangle.Rmd:
```diff
@@ -51,25 +50,30 @@ survey_data$sampling_week <- recode_factor(survey_data$sampling_week,
-survey_seines_2015_2018 <- hakaisalmon::survey_seines
-
+survey_seines_2015_2018 <- left_join(read_csv("../../jsp-data-packages/data/survey_data.csv"), read_csv("../../jsp-data-packages/data/seine_data.csv")) %>% 
+  left_join(read_csv("../../jsp-data-packages/data/sites.csv"))
```
```suggestion
  left_join(read_csv("https://raw.githubusercontent.com/HakaiInstitute/jsp-data/master/data/sites.csv"))
```

Br-Johnson commented on scripts/data_wrangle.Rmd:
```diff
@@ -140,25 +155,48 @@ fish_and_sealice_field_data_2019 <- left_join(survey_seines_fish_2019, sealice_f
-#### end 2019 update section
-sealice_lab_motiles <- hakaisalmon::sealice_lab_motiles
-sealice_field <- hakaisalmon::sealice_field
+fish_field_data <- read_csv("../../jsp-data-packages/data/fish_field_data.csv", guess_max = 10000)
```
```suggestion
fish_field_data <- read_csv("https://raw.githubusercontent.com/HakaiInstitute/jsp-data/master/data/fish_field_data.csv")
```

Br-Johnson reviewed (changes requested):
Looks good. Just change the absolute file paths to URLs of the raw data on the github repo. It looks like most of the sample sizes in sealice_summary_table increased. How did that happen?

jclgan committed:
Update scripts/data_wrangle.Rmd from absolute file path to GitHub URL

Co-Authored-By: Brett Johnson <brettthomasjohnson@gmail.com>

jclgan committed:
Update scripts/data_wrangle.Rmd from absolute file path to GitHub URL

Co-Authored-By: Brett Johnson <brettthomasjohnson@gmail.com>

jclgan committed:
Update scripts/data_wrangle.Rmd from absolute file path to GitHub URL

Co-Authored-By: Brett Johnson <brettthomasjohnson@gmail.com>

jclgan commented:
> It looks like most of the sample sizes in sealice_summary_table increased. How did that happen?

With the exception of 36 fish, it looks like none of the fish that were _field loused only_ were included in the original time series summary. Those 36 fish belong to three packages, which don't seem out of the ordinary upon inspection (within the date range, they received lab and/or field louse ID, and have corresponding records)

jclgan committed:
Re-ran setup & sealice chunks with updated data wrangling code

Re-added guess_max = 10000 as bound to read_csv "fish_field_data.csv" to ensure column types were properly recognized.

Br-Johnson commented on scripts/data_wrangle.Rmd:
````diff
@@ -11,7 +11,6 @@ A shiny app of these same data can be found [here](https://hecate.hakai.org/shin
 
 
 ```{r setup}
-library(hakaisalmon) #v0.2.1
````
We need to keep this package loaded because it contains the log_cumul_abund function

Br-Johnson reviewed (changes requested):
Just need to keep the hakaisalmon package loaded. Also, you should run the entire data_wrangle script and then run the entire report.Rmd to check the diff in the figures and statistics to see how your changes to data wrangle propagate through to the final product, to make sure there aren't any unintended consequences or breaking changes. You should be able to compare .png files in github to see how the figures changed
